model-optimizer/extensions/back/add_outputs_recursive.py

```diff
@@ -0,0 +1,319 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+import numpy as np
+
+from extensions.ops.If import If
+from extensions.ops.loop import Loop
+from extensions.ops.tensor_iterator import TensorIterator
+from mo.back.replacement import BackReplacementPattern
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_node_with_second_input
+from mo.graph.graph import Graph, Node
+from mo.ops.result import Result
+from mo.ops.unsqueeze import Unsqueeze
+
+
+def ti_find_iterations_count_for_output(step_node, output_rec):
+    def check_axis_end(record):
```

Comment:
why is the maximum returned? It should look minimum to be valid

Reply:
I can't agree. We have always equal number of iterations for inputs, but in theory if we have 2 inputs with different number of iterations we will iterate through both inputs. It would be strange if we ignore part of input with bigger number of iterations. 
Actually we can just add check that all inputs give the same iteration number - as we have in real cases.